beesdoo_product/models/beesdoo_product.py

```diff
@@ -390,6 +348,83 @@ def _inverse_purchase_price(self):
                     _("No Vendor defined for product '%s'") % product.name
                 )
 
+    def write(self, vals):
+        purchase_price = vals.get("purchase_price")
+        if purchase_price and purchase_price != self.purchase_price:
+            # We could write suggested_price to vals _now_, but the compute
+            # function is going to run regardless. It's a bit useless to run
+            # 'calculate_suggested_price' twice, but that's just how it is.
+            try:
+                suggested_price = self.calculate_suggested_price(
+                    price=purchase_price
+                )
+                self.adapt_list_price(vals, suggested_price=suggested_price)
+            except ValueError:
+                pass
+
+        super().write(vals)
+
+    @api.multi
+    def adapt_list_price(self, vals, suggested_price=None):
+        self.ensure_one()
+        if suggested_price is None:
+            suggested_price = self.suggested_price
+        vals.setdefault("list_price", suggested_price)
+
+    @api.multi
+    def calculate_suggested_price(self, price=None):
+        self.ensure_one()
+        suggested_price_reference = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("beesdoo_product.suggested_price_reference")
+        )
+        supplier = self._get_main_supplier_info()
+        if not supplier:
+            raise ValueError(
+                _("No supplier found for product {}").format(self.id)
+            )
```

Comment:
Should it really be considered an error if this error is catched and passed everytime the function is called ?

Reply:
Exceptions in Python are a little special. They're more frequently used as a program flow tool. At the end of every iteration, a `StopIteration` exception is thrown and caught by the interpreter to break a for-loop.

In this case, I need to convey to consumers of this function that no answer could be found. I could return None, but Python developers generally prefer catching an exception over checking against None.

Reply:
OK, thanks for the explanations :)